src/dmd/backend/tassert.h

```diff
@@ -24,7 +24,7 @@
 
 void util_assert(const char * , int) __attribute__((noreturn));
 
-static void local_assert(int line)
+__attribute__((noreturn)) static void local_assert(int line)
```

Comment:
Nit: put `__attribute__((noreturn))` at the end of the function declaration, for consistency with the one above.

Reply:
yes

Reply:
It doesn't work.

Reply:
How does it not work?

Reply:
I see, so you can put this attribute only at the end of function declarations, but with function definitions it can only be at start.
IIUC https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html#Attribute-Syntax attributes at the start of declaration are most likely to work.

Reply:
@WalterBright it doesn't compile as I showed in my previous comment https://github.com/dlang/dmd/pull/7498#discussion_r158579372.

Reply:
Yet an other reason to convert do D 😃.

Reply:
Nothing stopping you from declaring the function before defining it. 😉